packages/yumemi_lints/lib/src/command_services/update_command_service.dart

```diff
@@ -0,0 +1,186 @@
+import 'dart:io';
+
+import 'package:meta/meta.dart';
+import 'package:path/path.dart' as path;
+import 'package:pub_semver/pub_semver.dart';
+import 'package:yumemi_lints/src/models/exit_status.dart';
+import 'package:yumemi_lints/src/models/project_type.dart';
+
+class UpdateCommandService {
+  const UpdateCommandService();
+
+  ExitStatus call() {
+    try {
+      // Determine if it is a Dart project or a Flutter project
+      final projectType = _getProjectType();
+
+      switch (projectType) {
+        case ProjectType.dart:
+          return _updateDartProjectLintRule();
+        case ProjectType.flutter:
+          return _updateFlutterProjectLintRule();
+      }
+    } on Exception catch (e) {
+      print(e);
+      return ExitStatus.error;
+    }
+  }
+
+  ExitStatus _updateDartProjectLintRule() {
+    ProcessResult command;
+
+    // Determine if dvm is being used
+    final dvm = Directory(path.join(Directory.current.path, '.dvm'));
+    if (dvm.existsSync()) {
+      command = Process.runSync('dvm', ['dart', '--version']);
+    } else {
+      command = Process.runSync('dart', ['--version']);
```

Comment:
![ask-badge](https://img.shields.io/badge/review-ask-yellowgreen.svg) 
Instead of relying on the results of command executions, I thought it might be better to refer to the versions listed in the `pubspec.yaml` environment, as it would not depend on an individual's environment. What do you think? If it's a Flutter project but there's no specific version of Flutter mentioned, I'm thinking it might be best to not update and instead display an error message.

Reply:
I recognize that the version listed in pubspec.yaml is only a range of available versions.

For example, even if the version is listed as sdk: ^3.1.0, the project may be using v3.2.0.

Therefore, I use [dart --version] and [flutter --version] to get a clear version used by the project.

Reply:
@morikann 

> I recognize that the version listed in pubspec.yaml is only a range of available versions.
For example, even if the version is listed as sdk: ^3.1.0, the project may be using v3.2.0.

Your understanding is correct :+1:

However, with sdk: ^3.1.0, it might use v3.1.0. In that case, you would need to match the minimum available version rather than the new version.

I apologize if my understanding is incorrect 🙏